tests/unit/test_document_revision.py

```diff
@@ -0,0 +1,523 @@
+# pylint: disable=protected-access
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.cloud.documentai_toolbox.wrappers.document_revision import _OP_TYPE
+
+# try/except added for compatibility with python < 3.8
+try:
+    from unittest import mock
+except ImportError:  # pragma: NO COVER
+    import mock
+
+import pytest
+
+from google.cloud.documentai_toolbox import document_revision
+
+from google.cloud import documentai
+
+
+@pytest.fixture
+def get_revisions():
+    parent = document_revision.DocumentWithRevisions(document=None, revision_nodes=None)
+    second_parent = document_revision.DocumentWithRevisions(
+        document=None, revision_nodes=None
+    )
+    child = document_revision.DocumentWithRevisions(document=None, revision_nodes=None)
+    second_child = document_revision.DocumentWithRevisions(
+        document=None, revision_nodes=None
+    )
+    sub_child = document_revision.DocumentWithRevisions(
+        document=None, revision_nodes=None
+    )
+
+    parent.revision_id = 1
+    child.revision_id = 2
+    second_child.revision_id = 5
+    sub_child.revision_id = 3
+    second_parent.revision_id = 4
+
+    child.parent = parent
+    second_child.parent = parent
+    sub_child.parent = child
+
+    parent.root_revision_nodes = [parent, child, sub_child, second_parent, second_child]
+    child.root_revision_nodes = [parent, child, sub_child, second_parent, second_child]
+    sub_child.root_revision_nodes = [
+        parent,
+        child,
```

Comment:
Same comment as above. This is difficult to visualize what data is going where. Change this to initialize all values in the original object where you can. I've found that Bard/ChatGPT can do this quickly.

Reply:
`DocumentWithRevisions` doesn't take all those variables as parameters but i tried to simplify as much as i could.